lib/User/Manager.php

```diff
@@ -267,42 +267,32 @@ public function getCachedEntry($uid) {
 	}
 
 	/**
-	 * checks whether a user is still available on LDAP
+	 * Try to resolve missing DN by UUID
 	 *
 	 * @param string $dn
 	 * @return bool
-	 * @throws \Exception
-	 * @throws \OC\ServerNotAvailableException
 	 */
 	public function dnExistsOnLDAP($dn) {
-
-		//check if user really still exists by reading its entry
-		if (!\is_array($this->access->readAttribute($dn, '', $this->getConnection()->ldapUserFilter))) {
-			$lcr = $this->getConnection()->getConnectionResource();
-			if ($lcr === null) {
-				throw new \Exception('No LDAP Connection to server ' . $this->getConnection()->ldapHost);
-			}
-
-			try {
-				$uuid = $this->access->getUserMapper()->getUUIDByDN($dn);
-				if (!$uuid) {
-					return false;
-				}
+		try {
+			$uuid = $this->access->getUserMapper()->getUUIDByDN($dn);
+			if ($uuid) {
 				$newDn = $this->access->getUserDnByUuid($uuid);
 				//check if renamed user is still valid by reapplying the ldap filter
 				if (!\is_array($this->access->readAttribute($newDn, '', $this->getConnection()->ldapUserFilter))) {
 					return false;
 				}
 				$this->access->getUserMapper()->setDNbyUUID($newDn, $uuid);
 				return true;
-			} catch (\Exception $e) {
-				$this->logger->logException($e, ['app' => self::class]);
-				return false;
 			}
+		} catch (ServerNotAvailableException $e) {
+			// A. pass? Acc will deleted then.
+			// B. fail? in the middle of the user sync?
+		} catch (\OutOfBoundsException $e) {
+			// UUID attribute can't be detected this means that we need to drop the user
```

Comment:
It seems that OutOfBoundsExceptions can also be thrown if the `getUserDnByUuid` finds 2 or more users instead of only one. We'd need to decide what to do in that case

Reply:
If UUID is resolved into 2 different DNs it violates the requirement of uniqueness for UUID. 
As for me, we need to log a warning containing UUID and both DNs. Because this collision needs an action from LDAP administrator side.
In this case user should neither be deleted, nor remapped to any of  these new DNs, but old (missing) user DN should be left  "as is" until the collision is resolved.
@cdamken what do you think?

Reply:
@jvillafanez I chaged the implementation to throw `LengthException` when more than one DN have been found.

Reply:
We must take any action here, even if it's just logging the error, otherwise the admin won't notice the problem and it could cause issues that won't be possible to detect.

Reply:
missing DN does not look like a problem - user can be deleted by LDAP admin so this exception is expected for a valid reason. Added a log entry with `info` severity

Reply:
The UUID is unique and the only case that could 2 different DN's appears is that one is not valid anymore and update the one who is new.

Reply:
@cdamken "UUID is unique" is a requirement. But this requirement can be violated in the real life. Instead of just considering this situation as unbelievable I'd better add a piece of code to handle it explicitly. 

uuid attribute can be set to **any** ldap entry attribute  by the owncloud admin using `ldap_expert_uuid_user_attr`. Even to the attribute that have same values for **all** ldap users.

Dropping a user or random remapping is not a valid approach in this case. We should do nothing and log a warning about a duplication.